kivy/graphics/boxshadow.pyx

```diff
@@ -87,16 +91,36 @@ float sigmoid(float x) {
 
 void main (void){
 
-float distShadow = sigmoid(
-    roundedBoxSDF(
-        tex_coord0 * size - size/2.0,
-        size/2.0 - blur_radius * 1.5 - vec2(2.0),
-        border_radius
-    ) / (max(1.0, blur_radius) / 4.0)
-);
-
-vec4 shadow = vec4(1.0, 1.0, 1.0, clamp((1.0 - distShadow) * (frag_color.a * 2.0), 0.0, 1.0));
-gl_FragColor = shadow;
+    float adjustmentFactor = 0.0;
+    vec2 boxSize = size / 2.0;
+    vec2 boxPos = tex_coord0 * size - size/2.0;
+
+    if (inset == 1){
+        boxPos -= offset;
+        boxSize -= blur_radius + spread_radius;
+        adjustmentFactor = blur_radius;
+    }
+    else{
+        boxSize -= blur_radius * 1.5 + vec2(2.0);
+    }
+
+    float distShadow = sigmoid(
+        (
+            roundedBoxSDF(
+                boxPos,
+                boxSize,
+                border_radius - adjustmentFactor
+            ) - adjustmentFactor
+        ) / (max(1.0, blur_radius) / 4.0)
+    );
+
+    if (inset == 1){
+        gl_FragColor = vec4(1.0, 1.0, 1.0, distShadow);
+    }
+    else{
+        vec4 shadow = vec4(1.0, 1.0, 1.0, (1.0 - distShadow) * (frag_color.a * 2.0));
+        gl_FragColor = shadow;
```

Comment:
I guess (after the debugging session we had last month) that we can now avoid to assign the `vec4` to `shadow`, before assigning it to `gl_FragColor`?

Reply:
PS: I need to test it (deeply), but at a first glance the code looks good.
Expect a review in ~2 days.

Reply:
> I guess (after the debugging session we had last month) that we can now avoid to assign the `vec4` to `shadow`, before assigning it to `gl_FragColor`?

Correct! It also simplifies the logic.